Lib/test/test_capi/test_list.py

```diff
@@ -275,3 +275,39 @@ def test_list_astuple(self):
         self.assertRaises(SystemError, astuple, ())
         self.assertRaises(SystemError, astuple, object())
         self.assertRaises(SystemError, astuple, NULL)
+
+    def test_list_clear(self):
+        # Test PyList_Clear()
+        list_clear = _testcapi.list_clear
+
+        lst = [1, 2, 3]
+        self.assertEqual(list_clear(lst), 0)
+        self.assertEqual(lst, [])
+
+        lst = []
+        self.assertEqual(list_clear(lst), 0)
+        self.assertEqual(lst, [])
+
+        self.assertRaises(TypeError, list_clear, ())
+        self.assertRaises(TypeError, list_clear, object())
+
+        # CRASHES list_clear(NULL)
+
+    def test_list_extend(self):
```

Comment:
Add tests for `list_extend(lst, lst)`, `list_extend(lst, 'abc')` and `list_extend(lst, iter(...))`.